providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/in_process.py

```diff
@@ -27,18 +30,23 @@ class InProcessResolver:
     def __init__(self, config: Config, provider: AbstractProvider):
         self.config = config
         self.provider = provider
-        if not self.config.offline_flag_source_path:
-            raise ValueError(
-                "offline_flag_source_path must be provided when using in-process resolver"
+        self.flag_store = FlagStore(provider)
+        self.connector: FlagStateConnector = (
+            FileWatcher(
+                self.config.offline_flag_source_path,
+                self.provider,
+                self.flag_store,
+                self.config.offline_poll_interval_seconds,
             )
-        self.flag_store = FileWatcherFlagStore(
-            self.config.offline_flag_source_path,
-            self.provider,
-            self.config.offline_poll_interval_seconds,
+            if self.config.offline_flag_source_path
+            else GrpcWatcher(self.config, self.provider, self.flag_store)
         )
```

Comment:
Not a python expert, but does this if condition work correctly ?

Reply:
Might be hard to read from the multi-line formatting, but this is python's ternary operator, e.g.:
```my_var = True if x > 3 else False```

Functionally what this does is look at the `offline_flag_source_path` to determine whether the user wants to watch file system or grpc. I'd say this can be done better if we extend with more options in future, but with just 2 seemed sufficient to me

Reply:
ahh thanks, didn't know about this ternary operator :)